File: frontend/src/types/task.ts
// Erlaubte Werte für die Priorität.
export type Priority = "HIGH" | "MEDIUM" | "LOW";

// Erlaubte Zustände einer Aufgabe.
export type TaskStatus = "NEW" | "IN_PROGRESS" | "COMPLETED" | "DELETED";

// Beschreibt eine vollständige Aufgabe aus dem Backend.
export type Task = {
  id: number;
  title: string;
  description: string;
  priority: Priority;
  status: TaskStatus;
  dueDate: string;
};

// Enthält genau die Daten zum Erstellen einer Aufgabe.
export type CreateTaskInput = {
  title: string;
  description: string;
  priority: Priority;
  dueDate: string;
};

// Erlaubte Sortierparameter für das Backend.
export type TaskSort = "priority" | "dueDate";

File: backend/src/main/java/de/zaid/taskmanager/model/Task.java
package de.zaid.taskmanager.model;

import java.time.LocalDate;

// Enthält die Daten einer einzelnen Aufgabe.
public class Task {
    private final int id;
    private final String title;
    private final String description;
    private final Priority priority;
    private TaskStatus status;
    private final LocalDate dueDate;

    // Erstellt eine neue Aufgabe zuerst mit dem Status NEW.
    public Task(int id, String title, String description, Priority priority, LocalDate dueDate) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.priority = priority;
        this.status = TaskStatus.NEW;
        this.dueDate = dueDate;
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public Priority getPriority() {
        return priority;
    }

    public TaskStatus getStatus() {
        return status;
    }

    public LocalDate getDueDate() {
        return dueDate;
    }

    // Ändert den Status einer Aufgabe.
    public void setStatus(TaskStatus status) {
        this.status = status;
    }

}
